src/bob_request.rs

```diff
@@ -346,12 +346,16 @@ impl BobRequest {
 
     /// The transaction ID and output index of the zkapp used in the request.
     fn zkapp_outpoint(&self) -> Result<OutPoint> {
-        let txin = self
+        let outpoint = self
             .tx
             .input
-            .get(self.zkapp_input)
-            .context("the transaction ID that was passed in the request does not exist")?;
-        Ok(txin.previous_output)
+            .iter()
+            .find(|tx| tx.previous_output.txid == self.zkapp_tx.txid())
+            .context("the transaction ID that was passed in the request does not exist")?
```

Comment:
maybe a better error here would be "the zkapp transaction ID could not be found in the transaction's inputs"